boundary/analyze.py

```diff
@@ -263,6 +266,31 @@ def func_class_arithmetics(fns):
     fns.tainted = (fns.border | fns.insider | fns.sidecar) & fns.in_vmlinux
     fns.und = (fns.sched_outsider - fns.outsider_opt) | fns.border | fns.sidecar
 
+def get_func_decl_strs(signatures, fmt):
+    """Generate function declaration strings. If both strong and weak
+    symbol exist, keep only one.
+    """
+    decl_strs = set()
+    local_syms = set()
+
+    for fn in signatures:
+        (name, file) = fn
+        s = fmt.format(**decls[fn])
+
+        if file != global_fn_dict.get(name):
+            local_syms.add(name)
+
+        """If there are two global symbols, then one must be weak and
+        the other strong. Otherwise, original kernel fails to compile.
+        """
+        if s in decl_strs:
+            assert name not in local_syms, \
+                'Attempt to redirect a repeating local symbol %s' % str(fn)
+            continue
```

Comment:
This might look simpler, and does the same,

```python
        if s in decl_strs:
            assert name not in local_syms, \
                'Attempt to redirect a repeating local symbol %s' % str(fn)
            continue
```
→
```python
        assert (s in decl_strs) and (name not in local_syms), \
                'Attempt to redirect a repeating local symbol %s' % str(fn)
```
Note that `continue` is not necessary, because key can be added to a `set` multiple times. 

And `continue` doesn't benefit much performance either, because this is a very rare case.

Reply:
(s in decl_strs) is usually False